### ISSUE TITLE
[SCM-859] Subversion commands don't work if path or URL contains '@'

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1090]
**[Thomas Kunkel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=thomaskunkel)** opened **[SCM-859](https://issues.apache.org/jira/browse/SCM-859?redirect=false)** and commented

scm:update with the svn provider doesn't work when the working directory contains an "@"

Following the error message when the folder itself has an "@" sign

```java
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building svntest 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-scm-plugin:1.9.5:update (default-cli) @ svntest ---
[INFO] Executing: cmd.exe /X /C "svn --non-interactive update C:\SVNTest\failingTest@"
[INFO] Working directory: C:\SVNTest\failingTest@
[ERROR] Provider message:
[ERROR] The svn command failed.
[ERROR] Command output:
[ERROR] svn: E155007: None of the targets are working copies

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.357 s
[INFO] Finished at: 2018-02-07T12:35:08+01:00
[INFO] Final Memory: 11M/155M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-scm-plugin:1.9.5:update (default-cli) on project svntest: Command failed.The svn command failed. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Following the error when the "@" sign is in an parent folder name

```java
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building svntest 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-scm-plugin:1.9.5:update (default-cli) @ svntest ---
[INFO] Executing: cmd.exe /X /C "svn --non-interactive update C:\SVNTest\failingTest@\Subfolderfail"
[INFO] Working directory: C:\SVNTest\failingTest@\Subfolderfail
[ERROR] Provider message:
[ERROR] The svn command failed.
[ERROR] Command output:
[ERROR] svn: E200009: 'C:********@\Subfolderfail': a peg revision is not allowed here

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.248 s
[INFO] Finished at: 2018-02-07T12:35:33+01:00
[INFO] Final Memory: 11M/155M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-scm-plugin:1.9.5:update (default-cli) on project svntest: Command failed.The svn command failed. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

This error can be fixed by appending an additional "@" after the path in the command.

---

**Affects:** 1.9.5

**Remote Links:**
- [GitHub Pull Request #58
](https://github.com/apache/maven-scm/pull/58)
- [GitHub Pull Request #58
](https://github.com/apache/maven-scm/pull/58)
- [Subversion documentation
](https://stackoverflow.com/a/757510/696632)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Doesn't this also affect other commands?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This actually affects almost every command. As soon as an at sign is in the URL, the entire parser logic goes crazy and assumes this to be a username. Paired with `file://` it it will split with `:` and `@`.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [66a260a9107b6cec0c4ebfb9a42e9cc808512e90](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=66a260a9107b6cec0c4ebfb9a42e9cc808512e90).

I am certain that I may have missed some case, but let's face it only when a user reports an error.
